materialize-s3-parquet/parquet_data_converter.go

```diff
@@ -0,0 +1,275 @@
+package main
```

Comment:
There needs some explanations.
The basic idea of data converter is to convert the Flow data into an `Go` structure that can feed into the [write api](https://github.com/xitongsys/parquet-go/blob/master/writer/writer.go#L207) of the `parquet-go` lib. 

In their [example](https://github.com/xitongsys/parquet-go/blob/master/example/writer.go#L43-L52) the object is created using a pre-defined go struct. However, the Flow data schema is `open-ended`, which is defined by the catalog. So I feel it is natural to rely on `reflection` to construct the data object.

I agree that `parquet-go` is not the best option in terms of efficiency.  I did some research and did not see a good replacement. I left a [todo](https://github.com/estuary/connectors/blob/main/materialize-s3-parquet/parquet_file_processor.go#L21) to check back after ARROW complete its go-parquet package, and hope it will be better.

Reply:
If you want to meet about it, I'm happy to show you how to refactor the hot path at least (the set) to use type switches instead of reflection.

Reply:
Can you provide some general idea/approach that in your mind now?
I can look into that first.

Reply:
are you referring to sth. like this
https://golangdocs.com/reflection-type-switching-in-golang ?